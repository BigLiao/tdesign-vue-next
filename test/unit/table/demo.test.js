import { mount } from '@vue/test-utils';

// unit test for component in examples.
describe('Table', () => {
  it('base demo works fine', () => {
    const demo = require('@/examples/table/demos/base.vue').default;
    const wrapper = mount(demo);
    expect(wrapper.element).toMatchSnapshot();
  });
});
