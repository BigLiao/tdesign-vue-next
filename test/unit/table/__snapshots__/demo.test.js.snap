// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table base demo works fine 1`] = `
<div>
  <div
    class="t-table t-table--bordered t-table--striped t-table--hoverable t-size-s t-table-valign__top"
  >
    <div
      class="t-table-content"
      style="overflow: auto;"
    >
      <table
        style="table-layout: fixed;"
      >
        <colgroup>
          <col
            style="width: 100px; min-width: 100px;"
          />
          <col
            style="width: 200px; min-width: 200px;"
          />
          <col
            style="width: 100px; min-width: 100px;"
          />
          <col
            style="width: 200px; min-width: 200px;"
          />
          <col />
          <col
            style="width: 100px; min-width: 100px;"
          />
          <col
            style="width: 100px; min-width: 100px;"
          />
          <col
            style="width: 100px; min-width: 100px;"
          />
        </colgroup>
        <thead
          columnsprops="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
        >
          <tr>
            <td
              class="align-center text-ellipsis row"
              colspan="1"
              key="index"
              rowspan="1"
              style="overflow: hidden;"
            >
              index
            </td>
            <td
              class="align-left text-ellipsis row"
              colspan="1"
              key="type"
              rowspan="1"
              style="overflow: hidden;"
            >
              类型
            </td>
            <td
              class="align-left text-ellipsis test"
              colspan="1"
              key="platform"
              rowspan="1"
              style="overflow: hidden;"
            >
              平台
            </td>
            <td
              class="align-left text-ellipsis test2"
              colspan="1"
              key="property"
              rowspan="1"
              style="overflow: hidden;"
            >
              属性
            </td>
            <td
              class="align-left text-ellipsis test4"
              colspan="1"
              key="default"
              rowspan="1"
            >
              默认值
            </td>
            <td
              class="align-left text-ellipsis test3"
              colspan="1"
              key="needed"
              rowspan="1"
              style="overflow: hidden;"
            >
              是否必传
            </td>
            <td
              class="align-left text-ellipsis test3"
              colspan="1"
              key="detail.name"
              rowspan="1"
              style="overflow: hidden;"
            >
              详情信息
            </td>
            <td
              class="align-left text-ellipsis row"
              colspan="1"
              key="description"
              rowspan="1"
              style="overflow: hidden;"
            >
              说明
            </td>
          </tr>
        </thead>
        <tbody
          class="table-body"
        >
          <tr
            class=""
          >
            <td
              class="align-center text-ellipsis row"
              key="index"
              length="8"
              style="overflow: hidden;"
            >
              0
            </td>
            <td
              class="align-left text-ellipsis row"
              key="type"
              length="8"
              style="overflow: hidden;"
            >
              any[]
            </td>
            <td
              class="align-left text-ellipsis test"
              key="platform"
              length="8"
              style="overflow: hidden;"
            >
              公有
            </td>
            <td
              class="align-left text-ellipsis test2"
              key="property"
              length="8"
              style="overflow: hidden;"
            >
              data
            </td>
            <td
              class="align-left text-ellipsis test4"
              key="default"
              length="8"
            >
              []
            </td>
            <td
              class="align-left text-ellipsis test3"
              key="needed"
              length="8"
              style="overflow: hidden;"
            >
              Y
            </td>
            <td
              class="align-left text-ellipsis test3"
              key="detail.name"
              length="8"
              style="overflow: hidden;"
            >
              嵌套信息读取
            </td>
            <td
              class="align-left text-ellipsis row"
              key="description"
              length="8"
              style="overflow: hidden;"
            >
              数据源
            </td>
          </tr>
          <tr
            class=""
          >
            <td
              class="align-center text-ellipsis row"
              key="index"
              length="8"
              style="overflow: hidden;"
            >
              1
            </td>
            <td
              class="align-left text-ellipsis row"
              key="type"
              length="8"
              style="overflow: hidden;"
            >
              any[]
            </td>
            <td
              class="align-left text-ellipsis test"
              key="platform"
              length="8"
              style="overflow: hidden;"
            >
              公有
            </td>
            <td
              class="align-left text-ellipsis test2"
              key="property"
              length="8"
              style="overflow: hidden;"
            >
              data
            </td>
            <td
              class="align-left text-ellipsis test4"
              key="default"
              length="8"
            >
              []
            </td>
            <td
              class="align-left text-ellipsis test3"
              key="needed"
              length="8"
              style="overflow: hidden;"
            >
              Y
            </td>
            <td
              class="align-left text-ellipsis test3"
              key="detail.name"
              length="8"
              style="overflow: hidden;"
            >
              嵌套信息读取
            </td>
            <td
              class="align-left text-ellipsis row"
              key="description"
              length="8"
              style="overflow: hidden;"
            >
              数据源
            </td>
          </tr>
          <tr
            class=""
          >
            <td
              class="align-center text-ellipsis row"
              key="index"
              length="8"
              style="overflow: hidden;"
            >
              2
            </td>
            <td
              class="align-left text-ellipsis row"
              key="type"
              length="8"
              style="overflow: hidden;"
            >
              any[]
            </td>
            <td
              class="align-left text-ellipsis test"
              key="platform"
              length="8"
              style="overflow: hidden;"
            >
              公有
            </td>
            <td
              class="align-left text-ellipsis test2"
              key="property"
              length="8"
              style="overflow: hidden;"
            >
              data
            </td>
            <td
              class="align-left text-ellipsis test4"
              key="default"
              length="8"
            >
              []
            </td>
            <td
              class="align-left text-ellipsis test3"
              key="needed"
              length="8"
              style="overflow: hidden;"
            >
              Y
            </td>
            <td
              class="align-left text-ellipsis test3"
              key="detail.name"
              length="8"
              style="overflow: hidden;"
            >
              嵌套信息读取
            </td>
            <td
              class="align-left text-ellipsis row"
              key="description"
              length="8"
              style="overflow: hidden;"
            >
              数据源
            </td>
          </tr>
          <tr
            class=""
          >
            <td
              class="align-center text-ellipsis row"
              key="index"
              length="8"
              style="overflow: hidden;"
            >
              3
            </td>
            <td
              class="align-left text-ellipsis row"
              key="type"
              length="8"
              style="overflow: hidden;"
            >
              any[]
            </td>
            <td
              class="align-left text-ellipsis test"
              key="platform"
              length="8"
              style="overflow: hidden;"
            >
              公有
            </td>
            <td
              class="align-left text-ellipsis test2"
              key="property"
              length="8"
              style="overflow: hidden;"
            >
              data
            </td>
            <td
              class="align-left text-ellipsis test4"
              key="default"
              length="8"
            >
              []
            </td>
            <td
              class="align-left text-ellipsis test3"
              key="needed"
              length="8"
              style="overflow: hidden;"
            >
              Y
            </td>
            <td
              class="align-left text-ellipsis test3"
              key="detail.name"
              length="8"
              style="overflow: hidden;"
            >
              嵌套信息读取
            </td>
            <td
              class="align-left text-ellipsis row"
              key="description"
              length="8"
              style="overflow: hidden;"
            >
              数据源
            </td>
          </tr>
          <tr
            class=""
          >
            <td
              class="align-center text-ellipsis row"
              key="index"
              length="8"
              style="overflow: hidden;"
            >
              4
            </td>
            <td
              class="align-left text-ellipsis row"
              key="type"
              length="8"
              style="overflow: hidden;"
            >
              any[]
            </td>
            <td
              class="align-left text-ellipsis test"
              key="platform"
              length="8"
              style="overflow: hidden;"
            >
              公有
            </td>
            <td
              class="align-left text-ellipsis test2"
              key="property"
              length="8"
              style="overflow: hidden;"
            >
              data
            </td>
            <td
              class="align-left text-ellipsis test4"
              key="default"
              length="8"
            >
              []
            </td>
            <td
              class="align-left text-ellipsis test3"
              key="needed"
              length="8"
              style="overflow: hidden;"
            >
              Y
            </td>
            <td
              class="align-left text-ellipsis test3"
              key="detail.name"
              length="8"
              style="overflow: hidden;"
            >
              嵌套信息读取
            </td>
            <td
              class="align-left text-ellipsis row"
              key="description"
              length="8"
              style="overflow: hidden;"
            >
              数据源
            </td>
          </tr>
          <tr
            class=""
          >
            <td
              class="align-center text-ellipsis row"
              key="index"
              length="8"
              style="overflow: hidden;"
            >
              5
            </td>
            <td
              class="align-left text-ellipsis row"
              key="type"
              length="8"
              style="overflow: hidden;"
            >
              any[]
            </td>
            <td
              class="align-left text-ellipsis test"
              key="platform"
              length="8"
              style="overflow: hidden;"
            >
              公有
            </td>
            <td
              class="align-left text-ellipsis test2"
              key="property"
              length="8"
              style="overflow: hidden;"
            >
              data
            </td>
            <td
              class="align-left text-ellipsis test4"
              key="default"
              length="8"
            >
              []
            </td>
            <td
              class="align-left text-ellipsis test3"
              key="needed"
              length="8"
              style="overflow: hidden;"
            >
              Y
            </td>
            <td
              class="align-left text-ellipsis test3"
              key="detail.name"
              length="8"
              style="overflow: hidden;"
            >
              嵌套信息读取
            </td>
            <td
              class="align-left text-ellipsis row"
              key="description"
              length="8"
              style="overflow: hidden;"
            >
              数据源
            </td>
          </tr>
          <tr
            class=""
          >
            <td
              class="align-center text-ellipsis row"
              key="index"
              length="8"
              style="overflow: hidden;"
            >
              6
            </td>
            <td
              class="align-left text-ellipsis row"
              key="type"
              length="8"
              style="overflow: hidden;"
            >
              any[]
            </td>
            <td
              class="align-left text-ellipsis test"
              key="platform"
              length="8"
              style="overflow: hidden;"
            >
              公有
            </td>
            <td
              class="align-left text-ellipsis test2"
              key="property"
              length="8"
              style="overflow: hidden;"
            >
              data
            </td>
            <td
              class="align-left text-ellipsis test4"
              key="default"
              length="8"
            >
              []
            </td>
            <td
              class="align-left text-ellipsis test3"
              key="needed"
              length="8"
              style="overflow: hidden;"
            >
              Y
            </td>
            <td
              class="align-left text-ellipsis test3"
              key="detail.name"
              length="8"
              style="overflow: hidden;"
            >
              嵌套信息读取
            </td>
            <td
              class="align-left text-ellipsis row"
              key="description"
              length="8"
              style="overflow: hidden;"
            >
              数据源
            </td>
          </tr>
          <tr
            class=""
          >
            <td
              class="align-center text-ellipsis row"
              key="index"
              length="8"
              style="overflow: hidden;"
            >
              7
            </td>
            <td
              class="align-left text-ellipsis row"
              key="type"
              length="8"
              style="overflow: hidden;"
            >
              any[]
            </td>
            <td
              class="align-left text-ellipsis test"
              key="platform"
              length="8"
              style="overflow: hidden;"
            >
              公有
            </td>
            <td
              class="align-left text-ellipsis test2"
              key="property"
              length="8"
              style="overflow: hidden;"
            >
              data
            </td>
            <td
              class="align-left text-ellipsis test4"
              key="default"
              length="8"
            >
              []
            </td>
            <td
              class="align-left text-ellipsis test3"
              key="needed"
              length="8"
              style="overflow: hidden;"
            >
              Y
            </td>
            <td
              class="align-left text-ellipsis test3"
              key="detail.name"
              length="8"
              style="overflow: hidden;"
            >
              嵌套信息读取
            </td>
            <td
              class="align-left text-ellipsis row"
              key="description"
              length="8"
              style="overflow: hidden;"
            >
              数据源
            </td>
          </tr>
          <tr
            class=""
          >
            <td
              class="align-center text-ellipsis row"
              key="index"
              length="8"
              style="overflow: hidden;"
            >
              8
            </td>
            <td
              class="align-left text-ellipsis row"
              key="type"
              length="8"
              style="overflow: hidden;"
            >
              any[]
            </td>
            <td
              class="align-left text-ellipsis test"
              key="platform"
              length="8"
              style="overflow: hidden;"
            >
              公有
            </td>
            <td
              class="align-left text-ellipsis test2"
              key="property"
              length="8"
              style="overflow: hidden;"
            >
              data
            </td>
            <td
              class="align-left text-ellipsis test4"
              key="default"
              length="8"
            >
              []
            </td>
            <td
              class="align-left text-ellipsis test3"
              key="needed"
              length="8"
              style="overflow: hidden;"
            >
              Y
            </td>
            <td
              class="align-left text-ellipsis test3"
              key="detail.name"
              length="8"
              style="overflow: hidden;"
            >
              嵌套信息读取
            </td>
            <td
              class="align-left text-ellipsis row"
              key="description"
              length="8"
              style="overflow: hidden;"
            >
              数据源
            </td>
          </tr>
          <tr
            class=""
          >
            <td
              class="align-center text-ellipsis row"
              key="index"
              length="8"
              style="overflow: hidden;"
            >
              9
            </td>
            <td
              class="align-left text-ellipsis row"
              key="type"
              length="8"
              style="overflow: hidden;"
            >
              any[]
            </td>
            <td
              class="align-left text-ellipsis test"
              key="platform"
              length="8"
              style="overflow: hidden;"
            >
              公有
            </td>
            <td
              class="align-left text-ellipsis test2"
              key="property"
              length="8"
              style="overflow: hidden;"
            >
              data
            </td>
            <td
              class="align-left text-ellipsis test4"
              key="default"
              length="8"
            >
              []
            </td>
            <td
              class="align-left text-ellipsis test3"
              key="needed"
              length="8"
              style="overflow: hidden;"
            >
              Y
            </td>
            <td
              class="align-left text-ellipsis test3"
              key="detail.name"
              length="8"
              style="overflow: hidden;"
            >
              嵌套信息读取
            </td>
            <td
              class="align-left text-ellipsis row"
              key="description"
              length="8"
              style="overflow: hidden;"
            >
              数据源
            </td>
          </tr>
        </tbody>
        <tfoot />
      </table>
    </div>
    <div
      class="t-table-pagination"
    >
      <div
        class="t-pagination t-size-m"
      >
        <div
          class="t-pagination__total"
        >
          共 120 项数据
        </div>
         
        <div
          class="t-pagination__select"
        >
          <div
            class="t-popup-reference"
            style="width: 100%; position: relative;"
          >
             
            <div
              class="t-select t-size-m"
            >
              <svg
                class="t-icon t-icon-undefined t-size-m t-select-left-icon"
                style="display: none;"
              >
                <use
                  xlink:href="#t-icon-undefined"
                />
              </svg>
               
              <span
                class="t-select-placeholder"
                style="display: none;"
              >
                 请选择
              </span>
               
              <input
                name="t-select"
                style="width: 0px;"
                type="hidden"
                value="10"
              />
                
              <span
                class="t-select-selectedSingle"
                title="10"
              >
                
        10
      
              </span>
               
              <div
                class="t-input t-size-m t-select-input"
                style="display: none;"
              >
                <input
                  class="t-input__inner"
                  placeholder="10"
                  type="text"
                />
              </div>
               
              <svg
                class="t-icon t-icon-chevron-down t-size-m t-select-right-icon"
              >
                <use
                  xlink:href="#t-icon-chevron-down"
                />
              </svg>
               
              <svg
                class="t-icon t-icon-close t-size-m t-select-right-icon"
                style="display: none;"
              >
                <use
                  xlink:href="#t-icon-close"
                />
              </svg>
               
              <svg
                class="t-icon t-icon-loading t-size-m t-select-right-icon"
                style="display: none;"
              >
                <use
                  xlink:href="#t-icon-loading"
                />
              </svg>
               
              <!---->
            </div>
             
            <div
              class="resize-sensor"
              dir="ltr"
              style="pointer-events: none; position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden; max-width: 100%;"
            >
              <div
                class="resize-sensor-expand"
                style="pointer-events: none; position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden; max-width: 100%;"
              >
                <div
                  style="position: absolute; left: 0px; top: 0px; transition: 0s;"
                />
              </div>
              <div
                class="resize-sensor-shrink"
                style="pointer-events: none; position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden; max-width: 100%;"
              >
                <div
                  style="position: absolute; left: 0px; top: 0px; transition: 0s; width: 200%; height: 200%;"
                />
              </div>
            </div>
          </div>
        </div>
         
        <div
          class="t-pagination__btn t-pagination__btn--prev t-is-disabled"
          disabled="disabled"
        >
          <svg
            class="t-icon t-icon-chevron-left"
            version="1.1"
            viewBox="0 0 1024 1024"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M266.45122048 489.6415328C253.98241280000002 502.1382944 254.00507328000003 522.37688704 266.50183488 534.84569536L266.50183488 534.84569536 641.5903552 909.0951936 686.8451904 863.8403584 334.4904192 512.2516633600001 686.8451904 160.83449728 641.5903552 115.57966336000001 267.2264192 488.89166336 266.50183488 489.59088959999997 266.47651392 489.6162112Z"
              fill-opacity=".9"
              fill-rule="evenodd"
            />
          </svg>
        </div>
         
        <ul
          class="t-pagination__pager"
        >
          <li
            class="t-pagination__number t-is-current"
          >
            1
          </li>
           
          <li
            class="t-pagination__number t-pagination__number--more"
            style="display: none;"
          >
            <svg
              class="t-icon t-icon-ellipsis"
              version="1.1"
              viewBox="0 0 1024 1024"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                fill="none"
                fill-rule="evenodd"
              >
                <path
                  d="M0 0H1024V1024H0V0z"
                />
                <path
                  d="M192 448C227.346224 448 256 476.653776 256 512 256 547.346224 227.346224 576 192 576 156.653776 576 128 547.346224 128 512 128 476.653776 156.653776 448 192 448ZM512 448C547.346224 448 576 476.653776 576 512 576 547.346224 547.346224 576 512 576 476.653776 576 448 547.346224 448 512 448 476.653776 476.653776 448 512 448ZM832 448C867.3462208 448 896 476.653776 896 512 896 547.346224 867.3462208 576 832 576 796.6537792 576 768 547.346224 768 512 768 476.653776 796.6537792 448 832 448Z"
                  fill="#000"
                />
              </g>
            </svg>
          </li>
           
          <li
            class="t-pagination__number"
          >
            
        2
      
          </li>
          <li
            class="t-pagination__number"
          >
            
        3
      
          </li>
          <li
            class="t-pagination__number"
          >
            
        4
      
          </li>
          <li
            class="t-pagination__number"
          >
            
        5
      
          </li>
           
          <li
            class="t-pagination__number t-pagination__number--more"
          >
            <svg
              class="t-icon t-icon-ellipsis"
              version="1.1"
              viewBox="0 0 1024 1024"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                fill="none"
                fill-rule="evenodd"
              >
                <path
                  d="M0 0H1024V1024H0V0z"
                />
                <path
                  d="M192 448C227.346224 448 256 476.653776 256 512 256 547.346224 227.346224 576 192 576 156.653776 576 128 547.346224 128 512 128 476.653776 156.653776 448 192 448ZM512 448C547.346224 448 576 476.653776 576 512 576 547.346224 547.346224 576 512 576 476.653776 576 448 547.346224 448 512 448 476.653776 476.653776 448 512 448ZM832 448C867.3462208 448 896 476.653776 896 512 896 547.346224 867.3462208 576 832 576 796.6537792 576 768 547.346224 768 512 768 476.653776 796.6537792 448 832 448Z"
                  fill="#000"
                />
              </g>
            </svg>
          </li>
           
          <li
            class="t-pagination__number"
          >
            12
          </li>
        </ul>
         
        <div
          class="t-pagination__btn t-pagination__btn--next"
        >
          <svg
            class="t-icon t-icon-chevron-right"
            version="1.1"
            viewBox="0 0 1024 1024"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M758.8081664000001 489.6127065600001C771.27697472 502.10946816 771.2543136 522.34806144 758.757552 534.8168691200001L758.757552 534.8168691200001 383.6690304000001 909.0663616 338.4141952 863.8115328 690.7689683200001 512.20006272 338.4141952 160.8056710400001 383.6690304 115.5508371200001 758.41696832 489.22406272 758.757552 489.56206336Z"
              fill-opacity=".9"
              fill-rule="evenodd"
            />
          </svg>
        </div>
         
        <!---->
      </div>
    </div>
  </div>
</div>
`;
